docs/authentication/session-management-overview.mdx

```diff
@@ -0,0 +1,101 @@
+
+### JWT Strategy with Refresh/Access Tokens
+
+In the context of session management, the central element is the JSON Web Token (JWT), which serves as the token that encapsulates and conveys the authentication and authorization information necessary for maintaining secure user sessions.
+
+#### Access Tokens:
+
+- Access tokens play a pivotal role in this strategy. These tokens are short-lived JWTs issued upon successful user authentication.
+
+- They contain crucial user details and permissions in the form of claims.
+
+- When a user interacts with the application, they present the access token in their requests. The server validates the access token to ascertain the user's identity and determine the extent of their authorized actions within the application.
+
+#### Refresh Tokens:
+
+- Refresh tokens complement access tokens within the JWT strategy. These tokens are designed to have a longer lifespan compared to access tokens.
+
+- Their primary purpose is to securely facilitate the renewal of access tokens without necessitating frequent user logins.
+
+- In cases where an access token expires, the client can utilize the refresh token to request a new access token without needing the user's credentials again.
+
+- It's important to emphasize that refresh tokens should be stored and handled with great care, as their exposure could lead to unauthorized access. They should not be accessible by client-side scripts or prone to XSS attacks.
+
+#### Why/Why Not to Use Local Storage with JWTs:
+
+- The JWT, as the token, should not be stored in local storage due to potential security vulnerabilities.
+
+- Local storage is susceptible to Cross-Site Scripting (XSS) attacks, which could lead to the theft of tokens by malicious scripts.
+
+- Furthermore, tokens residing in local storage tend to have a more extended lifespan, increasing the risk window for potential attacks.
+
+- To mitigate these risks, it is recommended to use HTTP-only cookies combined with the Secure flag, enhancing security and safeguarding against XSS attacks.
+
+### Session Cookie Strategy
+
+This strategy uses cookies to manage user sessions.
+
+#### Session Initialization:
+
+- Upon login, the server generates a unique session identifier (session ID) and stores session data on the server.
+
+#### Setting the Cookie:
+
+- The server sends a session cookie to the client containing the session ID.
+
+- The client's web browser stores this cookie locally.
+
+#### Sending the Cookie:
+
+- The client automatically includes the session cookie in each subsequent request.
+
+#### Server-Side Session Data:
+
+- The server associates incoming requests with the appropriate session data using the session ID.
+
+#### Expiry and Timeout:
+
+- Session cookies can have an expiration time and a timeout mechanism to handle inactivity.
```

Comment:
should a token revocation be initiated on signout?

Reply:
I thought about this as well, so I added a small section. The session cookie should be removed and the tokens should be revoked, for more security